lib/SIL/SILFunctionType.cpp

```diff
@@ -2496,6 +2456,46 @@ static CanType copyOptionalityFromDerivedToBase(TypeConverter &tc,
   return base;
 }
 
+// SWIFT_ENABLE_TENSORFLOW
+// TODO: Move this to a common location (SILFunctionType.cpp).
+static CanSILFunctionType
+normalizeAutoDiffLinearMapType(CanSILFunctionType assocFnType) {
+  SmallVector<SILParameterInfo, 4> parameters;
+  for (auto param : assocFnType->getParameters())
+    parameters.push_back({param.getType(),
+                          ParameterConvention::Indirect_In_Guaranteed});
+  SmallVector<SILResultInfo, 4> results;
+  for (auto result : assocFnType->getResults())
+    results.push_back({result.getType(), ResultConvention::Indirect});
+  return SILFunctionType::get(
+      assocFnType->getGenericSignature(), assocFnType->getExtInfo(),
+      assocFnType->getCoroutineKind(), assocFnType->getCalleeConvention(),
+      parameters, assocFnType->getYields(), results,
+      assocFnType->getOptionalErrorResult(), assocFnType->getASTContext(),
+      assocFnType->getWitnessMethodConformanceOrNone());
+}
+
+// SWIFT_ENABLE_TENSORFLOW
+// TODO: Move this to a common location (SILFunctionType.cpp).
```

Comment:
Add a comment.

Reply:
Added comment in https://github.com/apple/swift/pull/26709/commits/1093416710d66105e8dc09a6110aed839b535975.